### PR TITLE
Dramatically speed up FID computation using DataLoader for asynchronous data loading

### PR DESCRIPTION
**Pull request by Vermeille**
Hello,

By using Pytorch's data loading pipeline, we can benefit of parallel and asynchronous data loading.

With the same batch size, My FID computation times goes from ~18 mins to ~3 mins.

### CODE REVIEW

**Comment by AtlantixJJ** [comment]
Hi, why not use original `torchvision.datasets.ImageFolder`?

**Comment by Vermeille** [comment]
because it forces to have a subdirectory in your image directory (because it gets labels from subdirs) which is an unnecessary and unnatural constraint for the problem at hand

**Comment by AtlantixJJ** [comment]
okay, that's reasonable. My workaround is to wrap one more folder. It is quite convenient.

**Comment by Vermeille** [comment]
that works too, but is less convenient to work with :)

**Comment by mseitzer** [comment]
Outdated due to #47.
